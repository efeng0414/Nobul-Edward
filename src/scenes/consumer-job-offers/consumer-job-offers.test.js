import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { BrowserRouter } from "react-router-dom";
import { shape } from "prop-types";
import PropTypes from "prop-types";
import ConsumerJobOffersWrapped from "./container";
import { Provider } from "react-redux";

describe("ConsumerJobOffers", () => {
  const {
    WrappedComponent: ConsumerJobOffers
  } = ConsumerJobOffersWrapped.WrappedComponent.WrappedComponent;

  const resolvePromise = () => Promise.resolve();
  const containerProps = {
    intl: global.getCustomIntl({
      myPostings: "My postings string",
      proposals: "Postings string",
      learningCenter: "Learning center string",
      proposalsTitle: "Proposals string"
    }),
    match: { params: { jobId: "abc123", jobType: "buy" } },
    jobDetails: {},
    buyOffers: { offerDetail: { agentId: "agent456" } },
    sellOffers: { agentId: "abc123" },
    getJobDetail: () => {},
    getJobOffers: resolvePromise,
    getMultipleAgentsWithAvatar: () => {}
  };
  const containerJSX = <ConsumerJobOffers {...containerProps} />;
  const mockStore = {
    subscribe: () => {},
    dispatch: () => {},
    getState: () => ({
      breakpoints: {
        currentBreakPoint: ""
      },
      authentication: {
        isLoading: false,
        currentUser: {}
      }
    })
  };
  const router = {
    history: new BrowserRouter().history,
    route: {
      location: {},
      match: {}
    }
  };
  const contextObject = {
    context: { intl: containerProps.intl, router, store: mockStore },
    childContextTypes: {
      intl: intlShape.isRequired,
      router: shape({}),
      store: PropTypes.object
    }
  };

  describe("Container Snapshot", () => {
    const containerMount = global.Enzyme.shallow(containerJSX, contextObject);
    it("should match stored snapshot", () => {
      expect(containerMount).toMatchSnapshot();
    });
  });

  describe("ComponentDidMount", () => {
    const getJobOffersMock = jest.fn(
      () =>
        new Promise(resolve => {
          resolve();
        })
    );

    containerProps.getJobOffers = getJobOffersMock;
    const wrapper = Enzyme.shallow(
      <ConsumerJobOffers {...containerProps} />,
      global.getIntlContext({ intl: containerProps.intl })
    );
    const jobObj = {
      jobType: containerProps.match.params.jobType,
      jobId: containerProps.match.params.jobId
    };

    it("should call props.getJobOffers on Mount", () => {
      expect(getJobOffersMock).toHaveBeenCalledWith(jobObj);
    });

    it("should call getUniqueAgentIds wth buyOffers on mount", async () => {
      const offers = containerProps.buyOffers;
      const getUniqueAgentIdsMock = jest.fn();
      const p = Promise.resolve("validMockData");
      containerProps.getJobOffers = jest.fn(() => p);
      const wrapper = Enzyme.shallow(
        <ConsumerJobOffers {...containerProps} />,
        global.getIntlContext({ intl: containerProps.intl })
      );
      wrapper.instance().getUniqueAgentIds = getUniqueAgentIdsMock;
      await p;

      expect(getUniqueAgentIdsMock).toHaveBeenCalledWith(offers);
    });

    it("should call getUniqueAgentIds wth sellOffers on mount", async () => {
      const getUniqueAgentIdsMock = jest.fn();
      const p = Promise.resolve("validMockData");
      containerProps.getJobOffers = jest.fn(() => p);
      containerProps.sellOffers = { agentId: "abc123" };
      containerProps.match.params.jobType = "sell";
      const offers = containerProps.sellOffers;

      const wrapper = Enzyme.shallow(
        <ConsumerJobOffers {...containerProps} />,
        global.getIntlContext({ intl: containerProps.intl })
      );
      wrapper.instance().getUniqueAgentIds = getUniqueAgentIdsMock;
      await p;

      expect(getUniqueAgentIdsMock).toHaveBeenCalledWith(offers);
    });

    it("should call props.getMultipleAgentsWithAvatar on mount", async () => {
      const getMultipleAgentsWithAvatarMock = jest.fn();
      containerProps.match.params.jobType = "sell";

      containerProps.getMultipleAgentsWithAvatar = getMultipleAgentsWithAvatarMock;
      const p = Promise.resolve("validMockData");
      containerProps.getJobOffers = jest.fn(() => p);

      const wrapper = Enzyme.shallow(
        <ConsumerJobOffers {...containerProps} />,
        global.getIntlContext({ intl: containerProps.intl })
      );
      await p;

      expect(getMultipleAgentsWithAvatarMock).toHaveBeenCalledWith({
        agentIdArray: wrapper
          .instance()
          .getUniqueAgentIds(containerProps.sellOffers)
      });
    });
  });
});
