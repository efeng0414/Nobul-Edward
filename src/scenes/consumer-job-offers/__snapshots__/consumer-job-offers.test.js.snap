// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerJobOffers Container Snapshot should match stored snapshot 1`] = `
<div
  className="consumer-job-offers"
>
  <InjectIntl(MyDashboardMeta)
    titleKey="helmet.myDashboard.allProposals"
  />
  <Breadcrumb
    prefixCls="ant-breadcrumb"
    separator=">"
  >
    <BreadcrumbItem
      prefixCls="ant-breadcrumb"
      separator="/"
    >
      <Link
        replace={false}
        to="/my-dashboard/postings"
      >
        My postings string
      </Link>
    </BreadcrumbItem>
    <BreadcrumbItem
      prefixCls="ant-breadcrumb"
      separator="/"
    >
      <Link
        replace={false}
        to="/my-dashboard/postings/buy/abc123"
      />
    </BreadcrumbItem>
    <BreadcrumbItem
      prefixCls="ant-breadcrumb"
      separator="/"
    >
      Postings string
    </BreadcrumbItem>
  </Breadcrumb>
  <div
    className="consumer-job-offers-grid"
  >
    <div
      className="consumer-job-offers-grid-ad"
    >
      <ConsumerLearningCenter
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {
              "learningCenter": "Learning center string",
              "myPostings": "My postings string",
              "proposals": "Postings string",
              "proposalsTitle": "Proposals string",
            },
            "now": [Function],
            "textComponent": "span",
          }
        }
        title={
          <h6>
            Learning center string
          </h6>
        }
      />
      <ConsumerStories
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {
              "learningCenter": "Learning center string",
              "myPostings": "My postings string",
              "proposals": "Postings string",
              "proposalsTitle": "Proposals string",
            },
            "now": [Function],
            "textComponent": "span",
          }
        }
      />
    </div>
    <div
      className="consumer-job-offers-grid-offers"
    >
      <Connect(withRouter(ConsumerJobOffersList))
        displayCount={true}
        displayFilter={true}
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {
              "learningCenter": "Learning center string",
              "myPostings": "My postings string",
              "proposals": "Postings string",
              "proposalsTitle": "Proposals string",
            },
            "now": [Function],
            "textComponent": "span",
          }
        }
        jobId="abc123"
        jobType="buy"
        title="Proposals string"
      />
    </div>
  </div>
</div>
`;
